frontend-react/src/pages/submissions/SubmissionTable.tsx

```diff
@@ -2,43 +2,60 @@ import { useResource } from "rest-hooks";
 import { useEffect } from "react";
 
 import useFilterManager from "../../hooks/filters/UseFilterManager";
-import useCursorManager from "../../hooks/filters/UseCursorManager";
+import useCursorManager, {
+    CursorActionType,
+} from "../../hooks/filters/UseCursorManager";
 import SubmissionsResource from "../../resources/SubmissionsResource";
 import { getStoredOrg } from "../../contexts/SessionStorageTools";
 import Table, { ColumnConfig, TableConfig } from "../../components/Table/Table";
-
-import SubmissionFilters from "./SubmissionFilters";
-
-/* TODO: This component needs to be able to sort ASC and DESC
- *   while being properly paginated. */
+import TableFilters from "../../components/Table/TableFilters";
 
 function SubmissionTable() {
     const filterManager = useFilterManager();
-    const cursorManager = useCursorManager(
-        filterManager.range.startRange.toISOString()
-    ); // First cursor set to StartRange on load
+    const {
+        cursors,
+        hasPrev,
+        hasNext,
+        update: updateCursors,
+    } = useCursorManager(filterManager.rangeSettings.start);
 
-    /* Our API call! Updates when any of the given state variables update */
+    /* Our API call! Updates when any of the given state variables update.
+     * The logical swap of cursors and range value is to account for which end of the
+     * range needs to update when paginating with a specific sort order.
+     *
+     * DESC -> Start [ -> ] End (Start uses cursor to increment towards end)
+     * ASC -> Start [ <- ] End (End uses cursor to increment towards start)
+     */
     const submissions: SubmissionsResource[] = useResource(
         SubmissionsResource.list(),
         {
             organization: getStoredOrg(),
-            cursor: cursorManager.values.cursor,
-            endCursor: filterManager.range.endRange.toISOString(),
-            pageSize: filterManager.pageSize.count + 1, // Pulls +1 to check for next page
-            sort: filterManager.sort.order,
+            cursor:
+                filterManager.sortSettings.order === "DESC"
+                    ? cursors.current
+                    : filterManager.rangeSettings.start,
+            endCursor:
+                filterManager.sortSettings.order === "ASC"
+                    ? cursors.current
+                    : filterManager.rangeSettings.end,
+            pageSize: filterManager.pageSettings.size + 1, // Pulls +1 to check for next page
```

Comment:
All this all code (cursor, endCursor, pageSize) logic feels like it should be abstracted elsewhere... but it's fine here. 
It's more of a surprise this level of detail is required at this higher level is all.

Reply:
I found the logic around using the cursor to update either start range or end range based on sort order very difficult to generalize because all those values are reactive and it became a nightmare. You'd essentially have an effect run whenever and one of those updated; startRange, endRange, cursor, or sortOrder. I thought reducers would solve a bit of this but it was still comparing so much state that the infinite refresh trap was wide open and I fell in.

Reply:
Maybe a `useCallback` that is only dependent on sort order and current cursor state and takes in a single value (i.e. the start or end range value) to fall back on would be a way to abstract that out and reuse it in future API calls where we need to pass in `cursorOrFallback()`! I'll try this out!